### PR TITLE
Relax run_uuid validation

### PR DESCRIPTION
**Pull request by eddiestudies**
# Relax the validation of run_uuid to allow for more id schemas
Update run_uuid regex to: _RUN_ID_REGEX = re.compile(r"^[a-zA-Z0-9][\w\-]{0,255}$")
from: _RUN_ID_REGEX = re.compile(r"^[0-9a-f]{32}$")
 
## This PR updates the regex and tests related to validation
Minor test edits related to failures for missing, unrelated packages were made

### CODE REVIEW

**Comment by eedeleon** [comment]
I do not think the failure is related to the PR:
tests/keras/test_keras_model_export.py::test_model_save_load FAILED

The test failed with:
 AssertionError: 
E       Arrays are not almost equal to 4 decimals
E       
E       Mismatch: 1.33%
E       Max absolute difference: 0.03125

This PR modified the run_uuid validation

File: tests/tracking/test_utils.py [hunk 1]
```diff
@@ -7,11 +8,18 @@
...
 from mlflow.store.rest_store import RestStore
-from mlflow.store.sqlalchemy_store import SqlAlchemyStore
+
 from mlflow.tracking.utils import _get_store, _TRACKING_URI_ENV_VAR, _TRACKING_USERNAME_ENV_VAR, \
     _TRACKING_PASSWORD_ENV_VAR, _TRACKING_TOKEN_ENV_VAR, _TRACKING_INSECURE_TLS_ENV_VAR, \
     get_db_profile_from_uri, _download_artifact_from_uri, TrackingStoreRegistry
 
+test_logger = logging.getLogger(__name__)
+
+try:
```
Comment (smurching) [inline tests/tracking/test_utils.py]: One question, is this change necessary & if not, could we remove it?

```
from mlflow.store.sqlalchemy_store import SqlAlchemyStore
```

seems to work for me

File: tests/helper_functions.py [hunk 2]
```diff
@@ -9,8 +10,14 @@
 
 import pandas as pd
 
-import mlflow.pyfunc.scoring_server as pyfunc_scoring_server
-import mlflow.pyfunc
+_logger = logging.getLogger(__name__)
```
Comment (smurching) [inline tests/helper_functions.py]: Similar comment to the SQLAlchemyStore one, can we remove this try/catch block?

File: tests/utils/test_validation.py [hunk 3]
```diff
@@ -44,9 +44,10 @@ def test_validate_tag_name():
 
 
 def test_validate_run_id():
-    for good_id in ["a" * 32, "f0" * 16, "abcdef0123456789" * 2]:
+    for good_id in ["a" * 32, "f0" * 16, "abcdef0123456789" * 2, "a" * 33, "a" * 31,
```
Comment (smurching) [inline tests/utils/test_validation.py]: Could be nice to test a length-255 string (like "a" * 255) and a string with non-hex alphanumeric characters (like "abcdefghijklmnopqrstuvwxyz") just to be super safe :)

File: mlflow/utils/validation.py [hunk 4]
```diff
@@ -13,7 +13,7 @@
 _VALID_PARAM_AND_METRIC_NAMES = re.compile(r"^[/\w.\- ]*$")
 
 # Regex for valid run IDs: must be a 32-character hex string.
```
Comment (smurching) [inline mlflow/utils/validation.py]: Micronit: would be nice to update this comment to reflect the updated validation, e.g.

"Regex for valid run IDs: must be an alphanumeric string of length 0 to 255" 

BTW, was it intentional to allow zero-length run IDs? Empty-string run IDs will break with the FileStore and IMO could be somewhat confusing for users to deal with - sorry if I missed discussion about this point elsewhere, please let me know if that's the case

**Review by smurching — COMMENTED** [review]
Thanks for the PR, left some comments! I don't have full context on the rationale behind this change, would you mind adding to the PR description and/or linking the appropriate GitHub issue in the PR description if one exists?

File: tests/tracking/test_utils.py (hunk 1 above)
Comment (eedeleon) [reply tests/tracking/test_utils.py]: I can remove it.

I added this to avoid importing sqlalchemy if I was not testing it since conftest loads this file and fails.

File: tests/helper_functions.py (hunk 2 above)
Comment (eedeleon) [reply tests/helper_functions.py]: sure thing, still not sold on all tests failing for missing imports but following the convention makes sense. 

I can pip install the referenced libraries

File: mlflow/utils/validation.py (hunk 4 above)
Comment (eedeleon) [reply mlflow/utils/validation.py]: zero-length run IDs are not supported, the first [] forces the first character to be alphanumeric, and is not optional. So minimum length is 1

Thanks for catching the stale comment!

File: tests/helper_functions.py (hunk 2 above)
Comment (smurching) [reply tests/helper_functions.py]: Ah for missing imports (I'm guessing `sqlalchemy`, amongst other packages, might have been missing?) running `pip install dev-requirements.txt` and `pip install test-requirements.txt` should install all dependencies needed to run tests.

**Review by smurching — APPROVED** [review]
LGTM, looks like that unrelated keras test failed again - retriggered the build & will merge once tests pass, thanks @eedeleon!

**Comment by smurching** [comment]
Hm Travis's build status seems off (shows as pending, but clicking "Details" you can see that the build passed: https://travis-ci.org/mlflow/mlflow/builds/511128282?utm_source=github_status&utm_medium=notification), merging to master :)

**Comment by akshaya-a** [comment]
thanks @smurching !
